library/src/main/java/com/novoda/downloadmanager/MigrationJob.java

```diff
@@ -53,6 +56,86 @@ public void run() {
         onUpdate(migrationStatus);
 
         Log.d(TAG, "about to extract migrations, time is " + System.nanoTime());
+        Log.d(TAG, "partial migrations are all EXTRACTED, time is " + System.nanoTime());
+
+        migratePartialDownloads(database, partialDownloadMigrationExtractor, downloadsPersistence);
+        migrateCompleteDownloads(migrationStatus, database, migrationExtractor, downloadsPersistence, internalFilePersistence);
+    }
+
+    private void onUpdate(MigrationStatus migrationStatus) {
+        for (MigrationCallback migrationCallback : migrationCallbacks) {
+            migrationCallback.onUpdate(migrationStatus);
+        }
+    }
+
+    private void migratePartialDownloads(SqlDatabaseWrapper database, PartialDownloadMigrationExtractor partialDownloadMigrationExtractor, DownloadsPersistence downloadsPersistence) {
```

Comment:
❓ QUESTION TIME! ❓ 

I'm wondering if we should have several jobs at this point and extract out some collaborators for these different migration steps.

We could have:
`PartialDownloadMigrationJob`
`CompleteDownloadMigrationJob`
`DeleteV1DatabaseJob`

My thinking is this might make the whole process easier to test. What do you think?

Reply:
We can try and tests for what we have first and let the pain points guide us towards a better more testable solution.

Reply:
In that case, I'll start testing these `MigrationJob`s now in a separate branch to see where that gets me 😄